app/interactors/fetch_exam_students.rb

```diff
@@ -0,0 +1,40 @@
+require 'interactors/kosapi_interactor'
+
+class FetchExamStudents < KOSapiInteractor
+
+  def setup(*args)
+    super
+    @people = {}
+  end
+
+  def perform(faculty_semester:)
+    client = kosapi_client(faculty_semester)
+
+    @exams = load_exam_events(faculty_semester).map do |exam|
+      students = fetch_exam_students(client, exam)
+      update_exam_students(exam, students)
+      exam
+    end
+  end
+
+  def results
+    { events: @exams, people: @people.values }
+  end
+
+  def load_exam_events(faculty_semester)
+    Event.where(event_type: 'exam', faculty: faculty_semester.faculty, semester: faculty_semester.code)#.where('starts_at > NOW()')
```

Comment:
This was supposed to be a disabled optimization. I've parametrized `FetchExamStudents` to allow enabling this as an option.